src/components/MovieListContent.ts

```diff
@@ -0,0 +1,70 @@
+import MovieList from '../domain/MovieList';
+import MovieItem from './MovieItem';
+import { Movie } from '../types/movie';
+import { $, $$ } from '../utils/domSelector';
+import InvalidMessage from './InvalidMessage';
+import { HTTPError } from '../api/HTTPError';
+import { MOVIE_MAX_COUNT } from '../constants';
+
+const MovieListContent = {
+  loadMovies: async () => {
+    $<HTMLButtonElement>('#more-button').classList.remove('hide');
+    MovieListContent.clear();
+    MovieListContent.loadMoreMovies();
+  },
```

Comment:
```suggestion
  async loadMovies() {
    $<HTMLButtonElement>('#more-button').classList.remove('hide');
    this.clear();
    this.loadMoreMovies();
  },
```

(제안)  자주 `MovieListContent`가 반복되는데, 계속 import 해온 모듈인가? 하면서 코드를 읽게 되네요 ㅠㅠ
 
 위와 같이 하거나 class를 써보는건 어떨까요? (ES6+ 니까!!)
 
 아니면 반대로 왜 이렇게 코드를 작성하는게 좋다고 생각하셨는지 애슐리의 생각을 들려주세요!

Reply:
처음에 객체로 작성한 이유는 페어랑 얘기하면서 굳이 클래스일 필요가 없다면 클래스로 작성할 필요가 없지 않을까라고 의견을 맞췄기 때문입니다. 그런데 저희가 제일 처음 작성한 컴포넌트가 `NavBar`여서 크게 클래스의 필요성을 느끼지 못했던 것 같습니다. 

조금 더 복잡한 컴포넌트들을 코딩하면서 동동이 말한 부분처럼 타입스크립트 객체에서는 `this`를 못 사용하고 계속 `MovieListContent`와 같이 다 적어줘야 하니까 조금 불편하다고 느꼈습니다. 거기에 컴포넌트들이 이름이 비슷하니까 더 한 번에 알아보기 힘들기도 했습니다. 그리고 반복적으로 같은 DOM 요소를 여러 메서드에서 셀렉팅하는 것도 비효율적이라고 느껴서 클래스 constructor에서 한 번만 셀렉팅 한 다음에 모든 메서드들이 사용하게 하는 것이 더 좋다고 느꼈습니다. 

그래서 리팩토링하면서 `NavBar`처럼 굳이 클래스일 필요가 없는 컴포넌트를 제외하고는 클래스로 변경해 봤습니다!